### DISCUSSION TITLE
OIDC token from github actions has no subject claim?

### DISCUSSION THREAD

**Discussion by avx-rvandegrift** [community/community#49966] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello,
I'm trying to use the Github Actions OIDC provider to get AWS credentials via https://github.com/aws-actions/configure-aws-credentials

I've got the AWS side setup, but my actions fail with:
> Error: Not authorized to perform sts:AssumeRoleWithWebIdentity
It looks like AWS doesn't like the token.

So I captured a token to troubleshoot further.  I tried to use it with awscli and got a bit more info:
```
$ aws sts assume-role-with-web-identity --role-arn $role_arn --role-session-name testing --web-identity-token file://token

An error occurred (InvalidIdentityToken) when calling the AssumeRoleWithWebIdentity operation: Missing a required claim: sub
```
Sure enough, the token I have from Github has no `sub` claim.  Instead, there's `oidc_sub` with the data that's documented for `sub`.  (i.e., it has `repo:<org>/<repo>:ref:refs/heads/main`)

What could be causing this, and how can I fix it?

**Comment by avx-rvandegrift** [comment, marked as answer]
I figured this out - AWS's error message is misleading.  I was generating my IAM conditions with terraform, and had a bug that resulted in an extra character in the `StringLike` check that's validating the subject.  So in fact, what was happening was my Condition was failing, and AWS was blaming it on the missing claim.

So: for other who might run into this, double check your conditions for any errors and ignore what AWS is saying :)

**Reply by bogdannazarenko** [reply]
@avx-rvandegrift were you able to perform Condition checks on other claims such as "actor" and not just "sub" claim?
```json
"StringLike": {
      "token.actions.githubusercontent.com:sub": "repo:owner/gha-test-abc:*",
      "token.actions.githubusercontent.com:actor": "myusername*"
    },
```

**Reply by avx-rvandegrift** [reply]
@bogdannazarenko looks like I have policies that check `aud` and `sub`, they seem to be working.  I haven't tried other claims.
